Lib/test/test_array.py

```diff
@@ -40,6 +40,12 @@ def test_bad_constructor(self):
         self.assertRaises(TypeError, array.array, 'xx')
         self.assertRaises(ValueError, array.array, 'x')
 
+    @support.cpython_only
+    def test_uninitialised_new(self):
+        # Prevent heap types from being created uninitialised (bpo-43916)
```

Comment:
I suggest to rename tests to "test_disallow_instantiation". "test_uninitialised_new" is not exactly the issue addressed by your change. Something like:

```suggestion
    def test_disallow_instantiation(self):
        # Ensure that the type disallows instantiation (bpo-43916)
```

Reply:
Yes, that looks better. Thanks!